// ink-box/dist.js
'use strict';

var _ink = require('ink');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _boxen = require('boxen');

var _boxen2 = _interopRequireDefault(_boxen);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Box = props => {
	const text = (0, _ink.renderToString)((0, _ink.h)(
		'span',
		null,
		props.children
	));
	return (0, _ink.h)(
		'span',
		null,
		(0, _boxen2.default)(text, props)
	);
};

Box.propTypes = {
	children: _propTypes2.default.any.isRequired,
	borderColor: _propTypes2.default.string,
	borderStyle: _propTypes2.default.oneOfType([_propTypes2.default.oneOf(['single', 'double', 'round', 'single-double', 'double-single', 'classic']), _propTypes2.default.shape({
		topLeft: _propTypes2.default.string,
		topRight: _propTypes2.default.string,
		bottomLeft: _propTypes2.default.string,
		bottomRight: _propTypes2.default.string,
		horizontal: _propTypes2.default.string,
		vertical: _propTypes2.default.string
	})]),
	dimBorder: _propTypes2.default.bool,
	padding: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.shape({
		top: _propTypes2.default.number,
		right: _propTypes2.default.number,
		bottom: _propTypes2.default.number,
		left: _propTypes2.default.number
	})]),
	margin: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.shape({
		top: _propTypes2.default.number,
		right: _propTypes2.default.number,
		bottom: _propTypes2.default.number,
		left: _propTypes2.default.number
	})]),
	float: _propTypes2.default.oneOf(['right', 'center', 'left']),
	backgroundColor: _propTypes2.default.string,
	align: _propTypes2.default.oneOf(['left', 'center', 'right'])
};

module.exports = Box;
